toonz/sources/toonz/shortcutpopup.cpp

```diff
@@ -415,4 +471,320 @@ void ShortcutPopup::onSearchTextChanged(const QString &text) {
+  QTextStream out(&file);
+  out << shortcutString;
+  file.close();
+  m_dialog->hide();
+}
+
+//-----------------------------------------------------------------------------
+
+void ShortcutPopup::onDeletePreset() {
+  // change this to 4 once RETAS shortcuts are updated
+  if (m_presetChoiceCB->currentIndex() <= 3) {
+    DVGui::MsgBox(DVGui::CRITICAL, tr("Included presets cannot be deleted."));
+    return;
+  }
+
+  QString question(tr("Are you sure you want to delete the preset: ") +
+                   m_presetChoiceCB->currentText() + tr("?"));
+  int ret = DVGui::MsgBox(question, QObject::tr("Yes"), QObject::tr("No"), 0);
+  if (ret == 0 || ret == 2) {
+    // cancel (or closed message box window)
+    return;
+  }
+  TFilePath presetDir =
+      ToonzFolder::getMyModuleDir() + TFilePath("shortcutpresets");
+  QString presetName = m_presetChoiceCB->currentText();
+  if (TSystem::doesExistFileOrLevel(presetDir +
+                                    TFilePath(presetName + ".ini"))) {
+    TSystem::deleteFile(presetDir + TFilePath(presetName + ".ini"));
+    buildPresets();
+    m_presetChoiceCB->setCurrentIndex(0);
+  }
+  if (Preferences::instance()->getShortcutPreset() == presetName)
+    setCurrentPresetPref("DELETED");
+  getCurrentPresetPref();
+}
+
+//-----------------------------------------------------------------------------
+
+void ShortcutPopup::importPreset() {
+  m_loadShortcutsPopup = new GenericLoadFilePopup("Load Shortcuts File");
+  m_loadShortcutsPopup->addFilterType("ini");
+  TFilePath shortcutPath = m_loadShortcutsPopup->getPath();
+  if (shortcutPath == TFilePath()) {
+    m_presetChoiceCB->setCurrentIndex(0);
+    return;
+  }
+  if (!showConfirmDialog()) return;
+
+  TFilePath presetDir =
+      ToonzFolder::getMyModuleDir() + TFilePath("shortcutpresets");
+  if (!TSystem::doesExistFileOrLevel(presetDir)) {
+    TSystem::mkDir(presetDir);
+  }
+  QString name        = shortcutPath.withoutParentDir().getQString();
+  std::string strName = name.toStdString();
+  if (TSystem::doesExistFileOrLevel(presetDir + TFilePath(name))) {
+    if (!showOverwriteDialog(name)) return;
+  }
+  showDialog("Importing Shortcuts");
+  TSystem::copyFile(presetDir + TFilePath(name), shortcutPath, true);
+  clearAllShortcuts(false);
+  setPresetShortcuts(presetDir + TFilePath(name));
+  return;
+}
+
+//-----------------------------------------------------------------------------
+
+void ShortcutPopup::onLoadPreset() {
+  QString preset = m_presetChoiceCB->currentText();
+  TFilePath presetDir =
+      ToonzFolder::getMyModuleDir() + TFilePath("shortcutpresets");
+  if (preset == "") return;
+  if (preset == "Load from file...") {
+    importPreset();
+    return;
+  }
+
+  if (!showConfirmDialog()) return;
+  showDialog("Setting Shortcuts");
+  if (preset == "OpenToonz") {
+    clearAllShortcuts(false);
+    TFilePath fp = TEnv::getConfigDir() + TFilePath("defopentoonz.ini");
+    setPresetShortcuts(fp);
+    return;
```

Comment:
For now, default shortcut preset files (defopentoonz.ini, otharmony.ini, etc.) are stored under `$TOONZCONFIG` directly.

How about moving them to some folder under `$TOONZPROFILES `? Because the user's shortcuts data are saved in `$TOONZPROFILES/layouts/settings.username/shotcuts.ini` and it would be clear to save them in the same folder.

I think `$TOONZPROFILES/layouts/defaultshortcutpresets` would be nice as the default room presets are also saved in `$TOONZPROFILES/layouts/rooms`. Maybe folder name can be shorter, but I have no idea...